cider/function/aggregate/CiderRuntimeFunctions.h

```diff
@@ -134,6 +135,15 @@ extern "C" ALWAYS_INLINE void cider_agg_id_proj_string_nullable(int8_t* str_data
   }
 }
 
+extern "C" int64_t look_up_string_id_from_hasher(const char* str_ptr,
+                                                 const int32_t str_len,
+                                                 const int64_t string_hasher_handle) {
+  auto string_hasher = reinterpret_cast<CiderStringHasher*>(string_hasher_handle);
+  int64_t id =
+      string_hasher->lookupIdByValue(CiderByteArray(str_len, (const uint8_t*)str_ptr));
```

Comment:
If the string is null, would this str_len be some really large random int?

Reply:
this depends on string extractor, for null string it should return {0, nullptr}, I think it's not necessary to do such check here.